23_Arithmetic2UnitTest/app_resources/testWorkgroup.comp.hlsl

```diff
@@ -24,6 +68,50 @@ struct DataProxy
     }
 };
 
+template<class Config, class Binop>
+struct PreloadedDataProxy
+{
+    using dtype_t = vector<uint32_t, Config::ItemsPerInvocation_0>;
+    static_assert(nbl::hlsl::is_same_v<dtype_t, type_t>);
+
+    NBL_CONSTEXPR_STATIC_INLINE uint32_t PreloadedDataCount = Config::VirtualWorkgroupSize / Config::WorkgroupSize;
+
+    template<typename AccessType>
+    void get(const uint32_t ix, NBL_REF_ARG(dtype_t) value)
+    {
+        value = preloaded[(ix-nbl::hlsl::workgroup::SubgroupContiguousIndex())>>Config::WorkgroupSizeLog2];
+    }
+    template<typename AccessType>
+    void set(const uint32_t ix, const dtype_t value)
+    {
+        preloaded[(ix-nbl::hlsl::workgroup::SubgroupContiguousIndex())>>Config::WorkgroupSizeLog2] = value;
```

Comment:
its expensive to call `SubgroupContiguousIndex`, ~~compute it and store it as a member (you'll test whether stateful accessors work properly with your code as well)~~

Actually because thw workgroup size is power of two, the difference between `ix` and your subgroup contiguous index doesn't contribute anything to the result after `>>` , you can just do `ix>>Config::WorkgroupSizeLog2` like the FFT sample does